Tests/Fixtures/TestProject/Project.xcodeproj/project.pbxproj

```diff
@@ -78,6 +79,7 @@
 		66C3C5E3C13325F351A3008F /* module.modulemap in CopyFiles */ = {isa = PBXBuildFile; fileRef = F2950763C4C568CC85021D18 /* module.modulemap */; };
 		6E8F8303759824631C8D9DA3 /* Localizable.strings in Resources */ = {isa = PBXBuildFile; fileRef = 9E17D598D98065767A04740F /* Localizable.strings */; };
 		7148A4172BFA1CC22E6ED5DB /* MainInterface.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 753001CDCEAA4C4E1AFF8E87 /* MainInterface.storyboard */; };
+		715DD7C4C194A04A5C4049EF /* ExternalTarget.framework in Embed Frameworks */ = {isa = PBXBuildFile; fileRef = 3DCDCD083AABD1D30EA34576 /* ExternalTarget.framework */; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
```

Comment:
Are there supposed to be 2 build files added? As these are referencing the same PBXFileReference, XcodeProj can't give them a deterministic id hence the failing diff tests. 
If they both need to be there then XcodeProj needs to be updated to take into account something that is different about them like the `settings`. That has to be done here https://github.com/tuist/XcodeProj/blob/master/Sources/XcodeProj/Utils/ReferenceGenerator.swift#L290

Reply:
Yeah, one is for linking and the other is to embed it. I notice this is also the case for Framework2.framework in that test project, but it does not cause a failing diff test. Any idea why?

Reply:
This will be fixed in https://github.com/tuist/XcodeProj/pull/518. Will update this PR once that is merged.